token/core/zkatdlog/nogh/loaders.go

```diff
@@ -48,12 +50,19 @@ type VaultTokenLoader struct {
 	TokenVault TokenVault
 }
 
+// LoadTokens takes an array of token identifiers (txID, index) and returns the keys in the vault
+// matching the token identifiers, the corresponding zkatdlog tokens, the information of the
+// tokens in clear text and the identities of their owners
+// LoadToken returns an error in case of failure
 func (s *VaultTokenLoader) LoadTokens(ids []*token3.ID) ([]string, []*token.Token, []*token.TokenInformation, []view.Identity, error) {
 	var tokens []*token.Token
 	var inputIDs []string
 	var inputInf []*token.TokenInformation
 	var signerIds []view.Identity
-
+	if s.TokenVault == nil {
```

Comment:
what about adding a constructor and making VaultTokenLoader instead? We can avoid this check every time.